src/ctfe_bc.d

```diff
@@ -1121,13 +1126,14 @@ Expression toExpression(const BCValue value, Type expressionType,
         auto offset = cast(uint)base;
         import ddmd.root.rmem : allocmemory;
 
-        auto resultString = cast(char*)allocmemory(length * sz);
+        auto resultString = cast(char*)allocmemory(length * sz + sz);
 
         foreach(i;0 .. length)
         {
             resultString[i] = cast(char) heapPtr._heap[offset];
             offset += sz;
```

Comment:
That doesn't look right. How do you represent UTF-16 and UTF-32 encodings in the uint[] heap?